Microsoft.Toolkit.Uwp.UI.Controls/ImageCropper/ImageCropper.cs

```diff
@@ -0,0 +1,373 @@
+            get
+            {
+                var realMinSelectSize = _imageTransform.TransformBounds(new Rect(default(Point), MinCropSize));
+                var minLength = Math.Min(realMinSelectSize.Width, realMinSelectSize.Height);
+                if (minLength < MinSelectedLength)
+                {
+                    var aspectRatio = KeepAspectRatio ? UsedAspectRatio : 1;
+                    var minSelectSize = new Size(MinSelectedLength, MinSelectedLength);
+                    if (aspectRatio >= 1)
+                    {
+                        minSelectSize.Width = minSelectSize.Height * aspectRatio;
+                    }
+                    else
+                    {
+                        minSelectSize.Height = minSelectSize.Width / aspectRatio;
+                    }
+
+                    return minSelectSize;
+                }
+
+                return new Size(realMinSelectSize.Width, realMinSelectSize.Height);
+            }
+        }
+
+        /// <inheritdoc/>
+        protected override void OnApplyTemplate()
+        {
+            UnhookEvents();
+            _layoutGrid = GetTemplateChild(LayoutGridName) as Grid;
+            _imageCanvas = GetTemplateChild(ImageCanvasPartName) as Canvas;
+            _sourceImage = GetTemplateChild(SourceImagePartName) as Image;
+            _maskAreaPath = GetTemplateChild(MaskAreaPathPartName) as Path;
+            _topButton = GetTemplateChild(TopButtonPartName) as Button;
+            _bottomButton = GetTemplateChild(BottomButtonPartName) as Button;
+            _leftButton = GetTemplateChild(LeftButtonPartName) as Button;
+            _rigthButton = GetTemplateChild(RightButtonPartName) as Button;
+            _upperLeftButton = GetTemplateChild(UpperLeftButtonPartName) as Button;
+            _upperRightButton = GetTemplateChild(UpperRightButtonPartName) as Button;
+            _lowerLeftButton = GetTemplateChild(LowerLeftButtonPartName) as Button;
+            _lowerRigthButton = GetTemplateChild(LowerRightButtonPartName) as Button;
+            HookUpEvents();
+            UpdateControlButtonVisibility();
+        }
+
+        private void HookUpEvents()
+        {
+            if (_imageCanvas != null)
+            {
+                _imageCanvas.SizeChanged += ImageCanvas_SizeChanged;
+            }
+
+            if (_sourceImage != null)
+            {
+                _sourceImage.RenderTransform = _imageTransform;
+                _sourceImage.ManipulationMode = ManipulationModes.TranslateX | ManipulationModes.TranslateY;
+                _sourceImage.ManipulationDelta += SourceImage_ManipulationDelta;
+            }
+
+            if (_maskAreaPath != null)
+            {
+                _maskAreaPath.Data = _maskAreaGeometryGroup;
+            }
+
+            if (_topButton != null)
+            {
+                _topButton.ManipulationMode = ManipulationModes.TranslateX | ManipulationModes.TranslateY;
+                _topButton.Tag = DragPosition.Top;
```

Comment:
Should use an attached property that you can strongly type vs. a Tag

Reply:
Then I wonder if it'd make sense to just use the property in XAML and have this code look for ALL 'buttons' underneath and register events to them if they have the attached property to be identified by, would condense this quite a bit.

Reply:
Maybe we should use Thumb or custom controls?Let me think about that.